tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py

```diff
@@ -0,0 +1,136 @@
+
+        # 2: call the function to create the mvar model
+        adjmat = shared_mvarmodel.mvaradjacencymatrix(eegwin)
+        
+        tempfilename = os.path.join(shared_tempdir, 'temp_'+str(win)+'.npz')
+        try:
+            np.savez_compressed(tempfilename, adjmat=adjmat)
+        except:
+            sys.stdout.write(traceback.format_exc())
+            return 0
+        # print("finished mvar job at ", win)
+        # sys.stdout.flush()
+    return 1
+
+def mvarjobwin(win):
+    # 1: fill matrix of all channels' next EEG data over window
+    eegwin = shared_raweeg[:, (shared_timepoints[win,0]) : (shared_timepoints[win,1]+1)]
+    # normalize eeg
+    # eegwin = _normalizets(eegwin)
+    # eegwin = (eegwin - np.mean(eegwin, axis=1)[:, np.newaxis]) / np.std(eegwin, axis=1)[:, np.newaxis]
+
+    # 2: call the function to create the mvar model
+    adjmat = shared_mvarmodel.mvaradjacencymatrix(eegwin)
+        
+    tempfilename = os.path.join(shared_tempdir, 'temp_'+str(win)+'.npz')
+    try:
+        np.savez_compressed(tempfilename, adjmat=adjmat)
+    except:
+        sys.stdout.write(traceback.format_exc())
+        return 0
+    return 1
+
+class ParallelMvar(BaseWindowModel):
+    def __init__(self, winsizems=250, stepsizems=125, samplerate=1000,numcores=None):
+        BaseWindowModel.__init__(self, 
+                                winsizems=winsizems, 
+                                stepsizems=stepsizems, 
+                                samplerate=samplerate)
+        if numcores is None:
+            self.numcores = psutil.cpu_count() - 1 
+        else:
+            self.numcores = numcores
+        # instantiate the mvar model
+        self.mvarmodel = MvarModel(self.winsize, self.stepsize)
+        # initialize results queue to store all results
+        # self.resultsqueue = mp.Queue()
+
+    def settempdir(self, tempdir=None):
+        self.tempdir = tempdir
+
+        if tempdir is None:
+            # set temporary directory
+            self.tempdir = tempfile.TemporaryDirectory()
+        print('created temporary directory', self.tempdir)
+
+    def runmvar(self, raweeg, listofwins=[]):
+        # get number of channels and samples in the raw data
+        numchans, numsignals  = raweeg.shape
+        if numchans >= 200:
+            warnings.warn("Whoa 200 channels to analyze? Could be too much to compute rn.")
+
+        # compute time and sample windows array
+        samplepoints = self.compute_samplepoints(numsignals, copy=False)
+        self.numwins = samplepoints.shape[0]
+
+        # Run multiprocessing pool over indices
+        with closing(mp.Pool(initializer=_initmvar, 
+                initargs=(raweeg, samplepoints, self.mvarmodel, self.tempdir))) as p:
+            mvarresults = p.map(mvarjobwin, range(0, self.numwins))
+        p.join()
+        # mvarresults = [1]
+        sys.stdout.write('Finished mvar model computation for ' +\
```

Comment:
should this be a logging statement instead, e.g. 

`logger.debug('Finsih comp for %d windows', self.numwins)`